### PR TITLE
include /etc/squid/conf.d/*.conf

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

include /etc/squid/conf.d/*.conf

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
https://github.com/uyuni-project/uyuni-docs/pull/3011

- [x] **DONE**

## Test coverage
- No tests
- [x] **DONE**

## Links
Issue(s): https://github.com/uyuni-project/uyuni-tools/issues/263
Depends on: https://github.com/uyuni-project/uyuni-tools/pull/320

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: containers/proxy-squid-image/Dockerfile [hunk 1]
```diff
@@ -30,6 +30,9 @@ COPY squid.conf /etc/squid/squid.conf
 RUN chown squid:squid /etc/squid/squid.conf
 RUN mkdir -p /run/squid && chown squid:squid /run/squid
 
+RUN mkdir -p /etc/squid/conf.d
+COPY squid_image.conf /etc/squid/conf.d/squid_image.conf
```
Comment (rjmateus) [inline containers/proxy-squid-image/Dockerfile:34]: the folder needs to have at least one file. This is why you are including this empty file?

**Review by agraul — APPROVED** [review]
LGTM, please make sure that you rebase https://github.com/uyuni-project/uyuni/pull/8773 after merging this PR (or merge the other PR first and rebase this one)

**Comment by mbussolotto** [comment]
@cbosdo podman is fine. I tested K3s running:
```
./mgrpxy install kubernetes config.tar.gz  --helm-proxy-chart oci://registry.opensuse.org/home/mbussolotto/branches/systemsmanagement/uyuni/master/charts/uyuni/proxy-helm --tuningSquid s_t.conf --tuningHttpd a_t.conf --logLevel=debug
```
where
```
uyuniproxyk3s:~ # cat s_t.conf
#MY SQUID CONF
uyuniproxyk3s:~ # cat a_t.conf
#MY APACHE CONF
```
The pod are running correctly and I have:
```
uyuniproxyk3s:~ # kubectl get configmap -o yaml
....
- apiVersion: v1
  data:
    apache_tuning: |
      # Add your Apache tuning configuration here
      #MY APACHE CONF
    config.yaml: |
...
    squid_tuning: |
      # Add your Squid tuning configuration here
      #MY SQUID CONF
  kind: ConfigMap
  metadata:
    annotations:
      meta.helm.sh/release-name: uyuni-proxy
      meta.helm.sh/release-namespace: default
    creationTimestamp: "2024-05-22T09:38:22Z"
    labels:
      app.kubernetes.io/managed-by: Helm
    name: proxy-configmap
    namespace: default
...
```
is it enough as kubernetes test?

File: containers/proxy-squid-image/squid_image.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+# squid_image.conf
```
Comment (cbosdo) [inline containers/proxy-squid-image/squid_image.conf:1]: Couldn't you add some useful comment explaining how to add squid config? That would surely be better than an empty file. And the filename could be something like `squid_example.conf` or so.

**Review by cbosdo — COMMENTED** [review]
LGTM, maybe the empty squid file could be more useful

File: containers/proxy-squid-image/uyuni-configure.py [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
-        file_content = re.sub(r"access_log .*", "access_log stdio:/proc/self/fd/1 squid", file_content)
+        file_content = re.sub(
+            r"cache_dir aufs .*",
+            f"cache_dir aufs /var/cache/squid {str(config['max_cache_size_mb'])} 16 256",
+            file_content,
+        )
+        file_content = re.sub(
+            r"access_log .*", "access_log stdio:/proc/self/fd/1 squid", file_content
+        )
+        file_content += "\n" + "include /etc/squid/conf.d/*" + "\n"
```
Comment (rjmateus) [inline containers/proxy-squid-image/uyuni-configure.py:24]: if we are always adding this line, why not to add it to the squid configuration directly?

Comment (mbussolotto) [reply containers/proxy-squid-image/uyuni-configure.py:24]: It has sense, but I guess the same is valid also for `access_log` param, am I right  ?
cc @cbosdo

Comment (cbosdo) [reply containers/proxy-squid-image/uyuni-configure.py:24]: The `access_log` would mess up the non-container install, so it has to stay here until we completely remove this installation method for Uyuni.

**Review by cbosdo — APPROVED** [review]
I still see Marek's changes for the linting. I suppose those are merged in `master`, you probably need to rebase on `master`. Everything else LGTM

File: containers/proxy-squid-image/uyuni-configure.py (hunk 3 above)
Comment (mbussolotto) [reply containers/proxy-squid-image/uyuni-configure.py:24]: right, so the same is valid also for this `include`. I'll keep the change here

**Comment by admd** [comment]
@mbussolotto please rebase now as other PR has been merged!

**Review by rjmateus — APPROVED** [review]

**Pull request merged**